processor/transaction/processor.go

```diff
@@ -39,7 +39,8 @@ func (p *processor) Transform(buf []byte) []beat.Event {
 	err := json.Unmarshal(buf, &pa)
 	if err != nil {
 		logp.Err("Error while unmarshaling data: %v", err)
-		return nil
+		// Effectively drop the data
+		return []beat.Event{}
```

Comment:
@roncohen I would probably even prefer if Transform could also return an error, so logging and treating of the error would happen in the server. Like this we only do it in one place. Also it would make testing `Transform` simpler I would think.